### PR TITLE
doc: release process and detailed steps

### PR DESCRIPTION
**Pull request by neeraj-laad**
Signed-off-by Neeraj Laad <neeraj.laad@gmail.com>

documented the release steps

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,14 +1,59 @@
-# Stacks Release Process
+# Appsody Stacks Release Process
+A release in this repo represents a new version of a particular stack - denoted by the release tag. It will contain the following assets:
+- archives for all the templates for the released stack on [GitHub releases](https://github.com/appsody/stacks/releases)
+- latest indexes for each of the repos (`incubator`, `experimental`, `stable`) on [GitHub releases](https://github.com/appsody/stacks/releases)
+- updated docker hub images with appropriate tags on [dockerhub](https://hub.docker.com/u/appsody)
```
Comment (mhamwala) [inline RELEASE.md]: nit: we could change this to: ...appropriate tags on [Appsody dockerhub](https://hub.docker.com/u/appsody)

File: RELEASE.md [hunk 2]
```diff
@@ -1,14 +1,59 @@
...
 
-This document outlines the process for building, testing and releasing Appsody stacks. 
+## Steps to create a new stack release
+The Appsody stacks are made available by creating a tagged GitHub release. Follow these steps to create a new release of an Appsody stack:
 
-`appsody/stacks` contains multiple stacks repos such as `experimental`, `incubator` and `stable`. Each repo includes several stacks. The build process allows you to build and release each stack and the indexes for each stack repo. 
+### Trigger the release on GitHub
+1. Navigate to https://github.com/appsody/stacks/releases
+1. Click _Draft a new release_ button
```
Comment (mhamwala) [inline RELEASE.md]: nit: we could make this bold perhaps: **_Draft a new release_**

File: RELEASE.md [hunk 3]
```diff
@@ -1,14 +1,59 @@
...
+1. Ensure the target branch is __master__
+1. Define a tag in the format `<stack-id>-v<version>` (example: java-microprpfile-v0.2.4)
```
Comment (mhamwala) [inline RELEASE.md]: java-microprpfile-v0.2.4 -> java-microprofile-v0.2.4

File: RELEASE.md [hunk 4]
```diff
@@ -1,14 +1,59 @@
...
+1. Set a title in the format `<stack-id> v<version>` (example: java-microprofile v0.2.4)
+1. Describe the notable changes to the stack as release notes
+1. Tick _This is a pre-release_ checkbox
```
Comment (mhamwala) [inline RELEASE.md]: nit: we could make this bold perhaps: **_This is a pre-release_**

File: RELEASE.md [hunk 5]
```diff
@@ -1,14 +1,59 @@
...
+1. Click _Publish release_ button
```
Comment (mhamwala) [inline RELEASE.md]: nit: we could make this bold perhaps: **_Publish release_**

File: RELEASE.md [hunk 6]
```diff
@@ -1,14 +1,59 @@
...
 
-[Travis-CI](https://travis-ci.org) is configured to build and test the stacks and indexes on GitHub pull requests. When a GitHub release is created manually, the built and tested stacks and indexes are published on DockerHub and GitHub releases. 
+### Monitor the Travis build
+1. Watch the [Travis build](https://travis-ci.com/appsody/stacks) for the release and ensure it passes
+1. Check the release artefacts to ensure these all exist:
```
Comment (mhamwala) [inline RELEASE.md]: This section seems a little repetitive, perhaps we could create an assets section at the top and link up to that?

File: RELEASE.md [hunk 7]
```diff
@@ -1,14 +1,59 @@
...
+    - archives for all the templates for the released stack on [GitHub releases](https://github.com/appsody/stacks/releases)
+    - latest indexes for each of the repos (`incubator`, `experimental`, `stable`) on [GitHub releases](https://github.com/appsody/stacks/releases)
+    - updated docker hub images with appropriate tags on [dockerhub](https://hub.docker.com/u/appsody)
 
-## Build
+### Update the pre-release
+1. Navigate to https://github.com/appsody/stacks/releases
+1. Click on the title of the pre-release you have just created
+1. Click _Edit release_ button
```
Comment (mhamwala) [inline RELEASE.md]: nit: we could make this bold perhaps: **_Edit release_**

File: RELEASE.md [hunk 8]
```diff
@@ -1,14 +1,59 @@
...
+1. Untick _This is a pre-release_ checkbox
```
Comment (mhamwala) [inline RELEASE.md]: nit: we could make this bold perhaps: **_This is a pre-release_**

File: RELEASE.md [hunk 9]
```diff
@@ -1,14 +1,59 @@
...
+1. Click _Edit release_ button
```
Comment (mhamwala) [inline RELEASE.md]: This part should be: Click the **_Update pre-release_** button

File: RELEASE.md [hunk 10]
```diff
@@ -1,14 +1,59 @@
...
 
-The `build.sh` script is a wrapper created for convenience and covers the process of identifying which stacks need to be built, builds stack images and templates, generates all repo indexes and then tests the stacks and indexes locally. 
+### Update v1 Index
+1. For now we are manually updating the [old appsody stacks index](https://github.com/appsody/stacks/blob/master/index.yaml)
+1. Create a pull request to update this file and update the index manually for the updated stack. Ususally, we only need to update the `version` and `url` fields for the updated stack
```
Comment (mhamwala) [inline RELEASE.md]: Ususally **->** Usually

File: RELEASE.md [hunk 11]
```diff
@@ -1,14 +1,59 @@
...
+1. Submit the pull request
+1. Once the pull request is approved, merge the changes
+
+
+# Releasing Dependent Stacks
+There is no fixed release schedule for Appsody stacks. Stacks should be released as we merge pull requests in this repo.
+
+If a stack is consumed by other stacks, then we must also update the dependent stacks and release new versions of those. For example, a change to `nodejs` stack will need a new release of `nodejs`, followed by a new release of `nodejs-express`.
```
Comment (mhamwala) [inline RELEASE.md]: Possible reword suggestion:

If a stack is inherited from another stack, then we must also update the dependent stacks and release new versions for those. For example, a change to `nodejs` will need a new release of the `nodejs` stack, followed by a new release of `nodejs-express` and the `nodejs-functions` stack.

File: RELEASE.md [hunk 12]
```diff
@@ -1,14 +1,59 @@
...
+
+# Appsody Stacks Release Process - Technical Overview
+
+This document outlines the process for building, testing and releasing Appsody stacks.
+
+`appsody/stacks` contains multiple stacks repos such as `experimental`, `incubator` and `stable`. Each repo includes several stacks. The build process allows you to build and release each stack and the indexes for each stack repo.
+
+[Travis-CI](https://travis-ci.org) is configured to build and test the stacks and indexes on GitHub pull requests. When a GitHub release is created manually, the built and tested stacks and indexes are published on DockerHub and GitHub releases.
```
Comment (mhamwala) [inline RELEASE.md]: could we change this link to https://travis-ci.com, 
I encountered a couple of problems with the .org

File: RELEASE.md [hunk 13]
```diff
@@ -35,8 +80,8 @@ The script makes use of several Travis environment variables to determine which
 
 `test.sh` script is currently being implemented. It will run tests for each stack using the assets created by the `package.sh` script.
 
-## Release
+## Release Phase
 
- The release process is invoked when a GitHub release is created with a tag manually. The tag must follow the `<name>-v<version>` format, for example: `java-microprofile-v0.2.1`. Currently, we create a release for each stack version. This triggers a Travis that builds, tests and publishes the stack being released. 
+ The release process is invoked when a GitHub release is created with a tag manually. The tag must follow the `<name>-v<version>` format, for example: `java-microprofile-v0.2.1`.
```
Comment (mhamwala) [inline RELEASE.md]: I don't think we need this part anymore as we mention it at the top

File: RELEASE.md [hunk 14]
```diff
@@ -35,8 +80,8 @@ The script makes use of several Travis environment variables to determine which
...
 
-`release.sh` script iterates over the **STACKS_LIST** environment variable to determine which stack need to be released and then publishes the template archives for that stack as GitHub release assets and publishes the stack images on [Appsody dockerhub](https://hub.docker.com/u/appsody). It also publishes all repo indexes as GitHub release assets.
+`release.sh` script iterates over the **STACKS_LIST** environment variable to determine which stack need to be released and then publishes the template archives for that stack as GitHub release assets and publishes the stack images on [Appsody dockerhub](https://hub.docker.com/u/appsody). It also publishes all repo indexes as GitHub release assets.
```
Comment (mhamwala) [inline RELEASE.md]: Reword suggestion:

The `release.sh` script iterates over the **STACKS_LIST** environment variable to determine which stack(s) need to be released. Then it publishes the template archives for that stack as GitHub release assets and publishes the stack images on [Appsody dockerhub](https://hub.docker.com/u/appsody). It also publishes all repo indexes as GitHub release assets.

File: RELEASE.md (hunk 1 above)
Comment (neeraj-laad) [reply RELEASE.md]: fixed

File: RELEASE.md (hunk 3 above)
Comment (neeraj-laad) [reply RELEASE.md]: oops.. fixed

File: RELEASE.md (hunk 4 above)
Comment (neeraj-laad) [reply RELEASE.md]: done

File: RELEASE.md (hunk 5 above)
Comment (neeraj-laad) [reply RELEASE.md]: done

File: RELEASE.md (hunk 6 above)
Comment (neeraj-laad) [reply RELEASE.md]: I agree but I'd rather have it repeated to avoid release errors.

File: RELEASE.md (hunk 7 above)
Comment (neeraj-laad) [reply RELEASE.md]: done

File: RELEASE.md (hunk 9 above)
Comment (neeraj-laad) [reply RELEASE.md]: Yes.

File: RELEASE.md (hunk 8 above)
Comment (neeraj-laad) [reply RELEASE.md]: done

File: RELEASE.md (hunk 10 above)
Comment (neeraj-laad) [reply RELEASE.md]: fixed

File: RELEASE.md (hunk 11 above)
Comment (neeraj-laad) [reply RELEASE.md]: have reworded a bit, let me know what u think

File: RELEASE.md (hunk 12 above)
Comment (neeraj-laad) [reply RELEASE.md]: done

Comment (neeraj-laad) [reply RELEASE.md]: done

File: RELEASE.md (hunk 14 above)
Comment (neeraj-laad) [reply RELEASE.md]: updated

File: RELEASE.md (hunk 2 above)
Comment (neeraj-laad) [reply RELEASE.md]: done

File: RELEASE.md [hunk 15]
```diff
@@ -1,14 +1,59 @@
-# Stacks Release Process
+# Appsody Stacks Release Process
+A release in this repo represents a new version of a particular stack - denoted by the release tag. It will contain the following assets:
+- archives for all the templates for the released stack on [Appsody stacks GitHub releases](https://github.com/appsody/stacks/releases)
+- latest indexes for each of the repos (`incubator`, `experimental`, `stable`) on [Appsody stacks GitHub releases](https://github.com/appsody/stacks/releases)
+- updated docker hub images with appropriate tags on [Appsody dockerhub](https://hub.docker.com/u/appsody)
```
Comment (skoh7645) [inline RELEASE.md]: nit: docker hub -> DockerHub (consistency with line 53)

Comment (neeraj-laad) [reply RELEASE.md]: @skoh7645 this got merged do you want to submit a PR for this?

**Pull request merged**